org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/scopes/BuildScopeUnit.scala

```diff
@@ -0,0 +1,82 @@
+package org.scalaide.core.internal.project.scopes
+
+import scala.tools.nsc.Settings
+import org.eclipse.core.resources.IContainer
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IResource
+import org.eclipse.core.runtime.IProgressMonitor
+import org.eclipse.core.runtime.SubMonitor
+import org.eclipse.jdt.core.IJavaModelMarker
+import org.scalaide.core.IScalaProject
+import org.scalaide.core.SdtConstants
+import org.scalaide.core.internal.builder.BuildProblemMarker
+import org.scalaide.core.internal.builder.zinc.SbtAwareBuildManager
+import org.scalaide.core.internal.project.CompileScope
+import sbt.inc.IncOptions
+import org.scalaide.core.internal.builder.zinc.EclipseSbtBuildManager
+
+class BuildScopeUnit(val scope: CompileScope, val owningProject: IScalaProject, settings: Settings,
+  val dependentUnitInstances: Seq[BuildScopeUnit] = Seq.empty)
+    extends SbtAwareBuildManager {
+
+  private val delegate =
+    new EclipseSbtBuildManager(owningProject, settings, Some(owningProject.underlying.getFile(".cache-" + scope.name)),
+        addToClasspath, srcOutputs)
+
+  private def isSrcOutPossessor: PartialFunction[(IContainer, IContainer), Boolean] = {
+    case (src, out) if scope.isValidSourcePath(src.getProjectRelativePath) => true
```

Comment:
The name is confusing to me. Maybe `manages`, or `managesSourceFolder`? If I understand correctly, this method tests that this build scope "manages" the given pair of source and output directory.